mpi-job-files/parseCommand.py

```diff
@@ -37,62 +38,67 @@
-    modelExecutable = '_'.join([os.path.join(modelBinsDir, sys.argv[i+1]), "mpi"])
-
-    # KLW 16-01-2024 https://github.com/StatCan/openmpp/issues/51
-    if not os.path.isfile(modelExecutable):
-        exit()
-      
-    # Enter unique mpijob name:
-    mpiJobName = f"{sys.argv[i+1]}-{str(time()).replace('.', '-')}".lower()
-    manifest = manifest.replace("#<mpiJobName>", mpiJobName)
-
-    # Pass name to file for monitoring mpijob later:
-    with open("./etc/mpiJobName", "w") as jN:
-      jN.write(mpiJobName)
-    i += 2
-
-  # Number of replicas to create:
-  elif (i + 1 < len(sys.argv) and re.match("^-mpiNp$", sys.argv[i]) and re.match("^[0-9]+$", sys.argv[i+1])):
-    manifest = manifest.replace("#<numberOfReplicas>", f"{sys.argv[i+1]}")
-    manifest = manifest.replace("#<mpirunOption>", \
-      f"- -n\n{12*' '}- '{sys.argv[i+1]}'\n{12*' '}#<mpirunOption>")
-    i += 2
-
-  # mpirun bind-to option:
-  elif (i + 1 < len(sys.argv) and re.match("^--bind-to$", sys.argv[i]) \
-  and re.match("^(core|socket|none)$", sys.argv[i+1])):
-    manifest = manifest.replace("#<mpirunOption>", \
-      f"- --bind-to\n{12*' '}- {sys.argv[i+1]}\n{12*' '}#<mpirunOption>")
-    i += 2
-
-  # mpirun environment variable options:
-  elif (i + 1 < len(sys.argv) and re.match("^-x$", sys.argv[i]) \
-  and re.match("[a-zA-Z_][a-zA-Z0-9_]*=[^=]+$", sys.argv[i+1])):
-    manifest = manifest.replace("#<mpirunOption>", \
-      f"- -x\n{12*' '}- {sys.argv[i+1]}\n{12*' '}#<mpirunOption>")
-    i += 2
-
-  # OpenM options and arguments:
-  elif (i + 1 < len(sys.argv) and re.match("^-OpenM\.", sys.argv[i]) \
-  and re.match("[a-zA-Z0-9_/\.-]+", sys.argv[i+1])):
-    openmOptions.append(sys.argv[i])
-    openmOptions.append(sys.argv[i+1])
-    i += 2
-
-  # Unrecognized command line options:
-  else:
-    unrecognized += f"{sys.argv[i]}\n"
-    i += 1
-
-# Write any unrecognized options to file for debugging:
-with open("./etc/unrecognizedCmdLineOptions", "w") as u:
-  u.write(unrecognized)
+############################
+# KLW 2024-02-07 replace command line argument parsing
+#input string (command line arguments) handling
+# merge argv array into a single string
+inputArgsString = ' '.join(sys.argv)
+# split on ' -' instead of just space
+theArgs = inputArgsString.split(" -")
+
+argDict = {}
+
+#arg[0] is the command itself, so ignore
+for i in range(1, len(theArgs)):
+  argItem = theArgs[i].split() # defaults to spliting on space
+  argKey = "-" + argItem[0] # replace the - at the beginning
+  argVal = ""
+  for j in range(1,len(argItem)):
+    argVal = argVal + argItem[j] # combine the rest of the argItem, removing spaces
+    
+  #finally, add to the Dict
+  argDict[argKey] = argVal.strip()
+  #print(argKey, " ", argVal)
+
+# add the command for completeness
+argDict["command"] = theArgs[0]
+
+### Modify the manifest template
+mv = argDict["-modelName"]
+modelExecutable = '_'.join([os.path.join(modelBinsDir, argDict["-modelName"]), "mpi"])
```

Comment:
Even though indexing Maps is O(1), still think `mv` should be used instead of `argDict["-modelName"]`

Reply:
I will add a comment.  The mv = argDict was needed because argDict would not work inside the curly braces that are present at a few places (template replacement).  It took me a bit of time to determine where they would and would not work.